Code/PgSQL/rdkit/mol_op.c

```diff
@@ -222,9 +222,15 @@ Datum mol_inchi(PG_FUNCTION_ARGS) {
       searchMolCache(fcinfo->flinfo->fn_extra, fcinfo->flinfo->fn_mcxt,
                      PG_GETARG_DATUM(0), NULL, &mol, NULL);
   str = MolInchi(mol, opts);
-  res = pnstrdup(str, strlen(str));
-  free((void *)str);
-  PG_RETURN_CSTRING(res);
+  if (!str || *str == 0) {
+    free((void *)str);
```

Comment:
If the string is null we shouldn't be freeing it.
It's better to remove this line